lib/puppet/pops/parser/egrammar.ra

```diff
@@ -656,6 +656,7 @@ classname
   | WORD
   | CLASSREF
   | CLASS { error val[0], "'class' is not a valid classname" }
+  | STRING { error val[0], "classnames must be bare words, not strings" }
```

Comment:
This is not just for the name of a class - this rule accepts `Foo`, `Foo::Bar`, `foo`, `foo::bar`, `not-name-but-word` and it is used for classes, defines, functions, etc.
Suggest changing this to be more generic "A String was found where it is not accepted"